plugins/hls-overloaded-record-dot-plugin/src/Ide/Plugin/OverloadedRecordDot.hs

```diff
@@ -158,89 +170,65 @@ descriptor :: Recorder (WithPriority Log) -> PluginId
-                    _ -> Nothing
-            resolveData crsM =
-                case supportsResolve of
-                    Just True -> Just $ toJSON $ ORDRD (caDocId ^. L.uri) crsM
-                    _         -> Nothing
-            mkCodeAction crsD crsM  = InR CodeAction
+        let mkCodeAction (crsM, nse)  = InR CodeAction
                 { -- We pass the record selector to the title function, so that
                   -- we can have the name of the record selector in the title of
                   -- the codeAction. This allows the user can easily distinguish
                   -- between the different codeActions when using nested record
                   -- selectors, the disadvantage is we need to print out the
                   -- name of the record selector which will decrease performance
-                _title = mkCodeActionTitle exts crsM crsD
+                  _title = mkCodeActionTitle exts crsM nse
                 , _kind = Just CodeActionKind_RefactorRewrite
                 , _diagnostics = Nothing
                 , _isPreferred = Nothing
                 , _disabled = Nothing
-                , _edit = changes crsM crsD
+                , _edit = Nothing
                 , _command = Nothing
-                , _data_ = resolveData crsM
+                , _data_ = Just $ toJSON $ ORD pId $ toJSON $ ORDRD (caDocId ^. L.uri) crsM
                 }
-            actions = map (mkCodeAction crsDetails) (RangeMap.filterByRange caRange crsMap)
+            actions = map mkCodeAction (RangeMap.filterByRange caRange crsMap)
         pure $ InL actions
     where
-    mkCodeActionTitle :: [Extension] -> Int -> IntMap.IntMap RecordSelectorExpr-> Text
-    mkCodeActionTitle exts crsM crsD =
+    mkCodeActionTitle :: [Extension] -> Int -> HsExpr (GhcPass 'Renamed) -> Text
+    mkCodeActionTitle exts crsM se =
         if OverloadedRecordDot `elem` exts
             then title
             else title <> " (needs extension: OverloadedRecordDot)"
         where
-            title = "Convert `" <> name (IntMap.lookup crsM crsD) <> "` to record dot syntax"
-            name (Just (RecordSelectorExpr _ se _)) = printOutputable se
-            name _                                  = ""
+            title = "Convert `" <> printOutputable se <> "` to record dot syntax"
+
+mkWorkspaceEdit:: Uri -> Int -> IntMap.IntMap RecordSelectorExpr -> [Extension] -> NextPragmaInfo-> Maybe WorkspaceEdit
+mkWorkspaceEdit uri crsM crsD exts pragma =
+    Just $ WorkspaceEdit
+    { _changes = Just (Map.singleton uri (edits (IntMap.lookup crsM crsD)))
+    , _documentChanges = Nothing
+    , _changeAnnotations = Nothing}
+    where pragmaEdit =
+            if OverloadedRecordDot `elem` exts
+            then Nothing
+            else Just $ insertNewPragma pragma OverloadedRecordDot
+          edits (Just crs) = convertRecordSelectors crs : maybeToList pragmaEdit
+          edits _          = []
```

Comment:
This case shouldn't happen and is a bug, right? Can we arrange to log or fail here?